### PR TITLE
issue 1483

### PR DESCRIPTION
**Pull request by nmccready**
bug(_async): iterating over objects has been changed to only iterate over thier own properties and not prototypes.

issue /#1483

### CODE REVIEW

**Comment by nmccready** [comment]
@thezuck @cthrax

File: spec/coffee/directives/api/utils/_async.spec.coffee [hunk 1]
```diff
@@ -49,7 +49,7 @@ describe "_async", ->
     it "handle callback passes an index", (done) ->
```
Comment (nmccready) [inline spec/coffee/directives/api/utils/_async.spec.coffee]: Here down is the only changes

File: spec/coffee/directives/api/utils/_async.spec.coffee [hunk 2]
```diff
@@ -133,6 +133,25 @@ describe "_async", ->
             expect(test).toEqual(known)
             done()
 
+      it 'prototypes should not be called', (done) ->
+        class Dummy
```
Comment (nmccready) [inline spec/coffee/directives/api/utils/_async.spec.coffee]: specs to test issue 1483 which was correct

File: src/coffee/directives/api/utils/_async.coffee [hunk 3]
```diff
@@ -149,6 +149,8 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
     if angular.isArray collection
       array = collection
     else
+      collection = _.pick collection, (val, propName) ->
```
Comment (nmccready) [inline src/coffee/directives/api/utils/_async.coffee]: This is not exactly ideal as it is O(n) twice. But until I refactor this whole file it's not worth the hassle.

**Comment by cthrax** [comment]
LGTM. I'd merge it.

**Pull request merged**